sanic/request.py

```diff
@@ -349,56 +351,55 @@ def cookies(self):
                 self._cookies = {}
         return self._cookies
 
+    @property
+    def content_type(self):
+        return self.headers.get("Content-Type", DEFAULT_HTTP_CONTENT_TYPE)
+
+    @property
+    def match_info(self):
+        """return matched info after resolving route"""
+        return self.app.router.get(self)[2]
+
+    # Transport properties (obtained from local interface only)
+
     @property
     def ip(self):
         """
         :return: peer ip of the socket
         """
-        if not hasattr(self, "_socket"):
-            self._get_address()
-        return self._ip
+        return self.conn_info.client if self.conn_info else ""
 
     @property
     def port(self):
         """
         :return: peer port of the socket
         """
-        if not hasattr(self, "_socket"):
-            self._get_address()
-        return self._port
+        return self.conn_info.client_port if self.conn_info else 0
 
     @property
     def socket(self):
-        if not hasattr(self, "_socket"):
-            self._get_address()
-        return self._socket
-
-    def _get_address(self):
-        self._socket = self.transport.get_extra_info("peername") or (
-            None,
-            None,
-        )
-        self._ip = self._socket[0]
```

Comment:
Should we be concerned that we are no longer setting `self._ip` and `self._port`. Yes, they are "hidden" properties, but maybe we leave them in with a dep notice?

Reply:
These were just cache for the `request.ip` and `request.port` properties, right? I don't think anyone other than those property accessors should've been touching them, and if someone still did, it is righteous to have their programs crash.

Another open question is whether `remote_addr` should return the local peer address when no proxy is used (it returns None in current master and IIRC on this too). The reasoning for this is that a web service is interested on user IP, irregardless of whether the service is installed behind a proxy or not, and `request.remote_addr or request.ip` isn't very practical. Probably better left for another PR though.

Reply:
Agreed. Going to merge in and add to `20.6`. Save for another day.